diesel/src/pg/expression/helper_types.rs

```diff
@@ -407,10 +407,10 @@ pub type cardinality<A> = super::functions::cardinality<SqlTypeOf<A>, A>;
 /// Return type of [`array_fill(value,array)`](super::functions::array_fill())
 #[allow(non_camel_case_types)]
 #[cfg(feature = "postgres_backend")]
-pub type array_fill<E> = super::functions::array_fill<SqlTypeOf<E>, E, Array<Integer>>;
+pub type array_fill<E, A> = super::functions::array_fill<SqlTypeOf<E>, E, A>;
 
 /// Return type of [`array_fill_with_lower_bound(value,array,array)`](super::functions::array_fill_with_lower_bound())
 #[allow(non_camel_case_types)]
 #[cfg(feature = "postgres_backend")]
-pub type array_fill_with_lower_bound<E> =
-    super::functions::array_fill_with_lower_bound<SqlTypeOf<E>, E, Array<Integer>, Array<Integer>>;
+pub type array_fill_with_lower_bound<E, A1, A2> =
```

Comment:
Not sure why i have to write generics here for arguments that have concrete types, but this works. An explanation would be appreciated!

Reply:
`define_sql_function!` generates a function signature as follows for the `array_fill` function:

```rust
fn array_fill<E: SingleValue, value: T1, dim: T2>(value: T1, dim: T2) -> _
where T1: AsExpression<E>, 
            T2: AsExpression<Array<Integer>>
```

The additional `AsExpression` bounds enable diesel to accept a typed SQL expression (like for example a column) or a rust side value (for example `5`) as argument in this position.